### PR TITLE
Bump helm to v3.11.1 to address CVE-2023-25165

### PR DESCRIPTION
**Pull request by tmshort**
Requires a number of other dependency updates.

### CODE REVIEW

**Comment by tmshort** [comment]
Fixes OCPBUGS-7683

File: internal/uploadmgr/gc.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ func (gc *bundleGC) Start(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	bundleInformer.AddEventHandler(toolscache.ResourceEventHandlerFuncs{
+	_, _ = bundleInformer.AddEventHandler(toolscache.ResourceEventHandlerFuncs{
```
Comment (joelanford) [inline internal/uploadmgr/gc.go:42]: Hmm. I probably totally overlooked these return values originally. Anything we can/should do with either of these?

If not, it probably warrants a comment.

**Review by joelanford — APPROVED** [review]
A couple of nits, which would be fine to address in follow-ups.

File: internal/uploadmgr/gc.go (hunk 1 above)
Comment (tmshort) [reply internal/uploadmgr/gc.go:42]: Honestly, I was just trying to fix the lint without changing the behavior. But I could check the error.

File: test/e2e/plain_provisioner_test.go [hunk 2]
```diff
@@ -2243,7 +2243,7 @@ func checkProvisionerBundle(object client.Object, provisionerPodName string) err
 		return err
 	}
 
-	return exec.Stream(remotecommand.StreamOptions{
+	return exec.StreamWithContext(context.Background(), remotecommand.StreamOptions{
```
[the remark this answers is not in the document]
Comment (joelanford) [reply test/e2e/plain_provisioner_test.go]: Nit insofar as "I wouldn't block the PR on this" :)

**Comment by tmshort** [comment]
The `e2e/e2e-kind` test that runs `make e2e` is failing locally on both this branch and on `main`.

**Pull request merged**